saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/reference/TagModifyingJudgmentReferenceRemover.java

```diff
@@ -0,0 +1,128 @@
+package pl.edu.icm.saos.enrichment.reference;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+import pl.edu.icm.saos.common.json.JsonStringParser;
+import pl.edu.icm.saos.common.json.JsonStringWriter;
+import pl.edu.icm.saos.persistence.enrichment.EnrichmentTagRepository;
+import pl.edu.icm.saos.persistence.enrichment.model.EnrichmentTag;
+import pl.edu.icm.saos.persistence.repository.JudgmentRepository;
+
+/**
+ * Class for removing reference to judgment from {@link EnrichmentTag}.
+ * It supports tags where reference removing is based on modifying {@link EnrichmentTag#getValue()}.
```

Comment:
Abstract class for services that remove references to judgments from {@link EnrichmentTag}s.